src/compiler/types.rs

```diff
@@ -890,22 +941,40 @@ impl PropertyAssignment {
     }
 }
 
+pub trait HasTypeParametersInterface {
+    fn maybe_type_parameters(&self) -> Option<&NodeArray>;
```

Comment:
You could give this a default implementation of `None` if there are types that will never have this populated.

Reply:
Ya that's a good thing to keep in mind, there's been one or two places where it seemed like maybe a default trait method implementation might be useful